### ISSUE TITLE
IPC timed out after system is booted on BYT with sof master branch

### ISSUE THREAD

**Issue by keqiaozhang** [thesofproject/sof#293]
Failed to load tplg and ipc time out. Trace log is not available.
Test config: sof-dev + sof-master + soft-master
sof-dev: bafb590
sof-master: 9944905
soft-master: 41861aa
tplg name: test-ssp2-mclk-0-I2S-volume-s16le-s16le-48k-19200k-codec.tplg

[dmesg-byt.log](https://github.com/thesofproject/sof/files/2344565/dmesg-byt.log)

**Comment by jajanusz** [comment]
Commit https://github.com/thesofproject/sof/commit/9d552d532bb9f57c05a9dae0944057b2aaf0ec9b didn't touch BYT at all.

Please check if it worked after and before this one - https://github.com/thesofproject/sof/commit/35ed394a26a4ec70bc3b47970c7f9c00a57d113b

**Comment by keqiaozhang** [comment]
@jajanusz 
This regression is not introduced by your commit 9d552d5.
This issue happened several days ago after next branch merged into master.

**Comment by jajanusz** [comment]
Actually there is one thing that I'm not sure if is ok for baytrail (because I couldn't test it) - it's feature mask in rimage - https://github.com/thesofproject/sof/pull/290/commits/a53d982e1675238cce9db2ced6b8c00873799879 
So after you'll be sure that you resolved all issues related to master&next merge, that's the only thing that changed for baytrail.

**Comment by keqiaozhang** [comment]
I just tested it with 35ed39 (before a53d982) and this issue still can be reproduced. so this issue may not related to your changes. I will try to bisect tomorrow.

**Comment by lgirdwood** [comment]
@keqiaozhang pleas provide rmbox output too, this could be related to the list corruption which should be fixed tomorrow.

**Comment by keqiaozhang** [comment]
Bisection failed, because there're plenty of commits are unable to build.
@lgirdwood rmbox is not available after system is booted. Like shown below:
zql@zql-MB-T:~$ sudo rmbox -t
using 19.20MHz timestamp clock

**Comment by keqiaozhang** [comment]
@lgirdwood 
Checked with the latest code base, this issue is still there.
sof-master:4c0b1d4

**Comment by lgirdwood** [comment]
@keqiaozhang I mean rmbox without the -t

**Comment by keqiaozhang** [comment]
@lgirdwood 
Same result, there's no error log output.
zql@zql-MB-T:~$ sudo rmbox

Error log:
using 19.20MHz timestamp clock

**Comment by bardliao** [comment]
schedule_task_init and schedule_task_config are missing in byt-ipc.c.
Please try the patch below.

[0001-ipc-add-missing-schedule_task_init-config.patch.txt](https://github.com/thesofproject/sof/files/2363909/0001-ipc-add-missing-schedule_task_init-config.patch.txt)

**Comment by lgirdwood** [comment]
@bardliao pls send PR if it fixes issue.

**Comment by bardliao** [comment]
@lgirdwood Yes, the [PR 332](https://github.com/thesofproject/sof/pull/332) has been merged.

**Comment by lgirdwood** [comment]
@keqiaozhang can now be validated.

**Comment by mengdonglin** [comment]
@ZhendanYang please help to verify if this fix can work.

**Comment by ZhendanYang** [comment]
@lgirdwood @mengdonglin @bardliao 
Actually, I verified this issue yesterday and told the result to bard already. 
This issue was blocked by #274 , so I couldn't make sure this issue is fixed or not. SInce bard debuged this issue with a commit without issue #274 , so I may re-test this issue after #274 fixed.
[    8.349951] sof-audio sof-audio: error : DSP panic!
[    8.349990] sof-audio sof-audio: error: out of memory
[    8.349996] sof-audio sof-audio: error: trace point 00004170
[    8.350002] sof-audio sof-audio: error: DSP Firmware Oops
[    8.350008] sof-audio sof-audio: error: Exception Cause: IllegalInstructionCause, Illegal instruction
[    8.350016] sof-audio sof-audio: EXCCAUSE 0x00000000 EXCVADDR 0x00000000 PS       0x00060525 SAR     0x00000011
[    8.350024] sof-audio sof-audio: EPC1     0xff2ce4fd EPC2     0x00000000 EPC3     0x00000000 EPC4    0xff2c4adc
[    8.350031] sof-audio sof-audio: EPC5     0x00000000 EPC6     0x00000000 EPC7     0x00000000 DEPC    0x00000000
[    8.350039] sof-audio sof-audio: EPS2     0x00000000 EPS3     0x00000000 EPS4     0x00060720 EPS5    0x00000000
[    8.350046] sof-audio sof-audio: EPS6     0x00000000 EPS7     0x00000000 INTENABL 0x00008b48 INTERRU 0x000000a2
[    8.350054] sof-audio sof-audio: stack dump from 0xff327f20

**Comment by ZhendanYang** [comment]
Test with latest code base, this issue can't reproduce.
Test env:
sof master: 299ccbc
sof tool: 60c3937
kernel: sof-dev 51ebd5d
